packages/core/src/modules/mdoc/MdocDeviceResponse.ts

```diff
@@ -175,18 +185,51 @@ export class MdocDeviceResponse {
     const publicDeviceJwk = COSEKey.import(deviceKeyInfo.deviceKey).toJWK()
 
     const deviceResponseBuilder = await DeviceResponse.from(mdoc)
-      .usingPresentationDefinition(presentationDefinition)
       .usingSessionTranscriptForOID4VP(sessionTranscriptOptions)
       .authenticateWithSignature(publicDeviceJwk, 'ES256')
 
+    if (options.presentationDefinition) {
+      deviceResponseBuilder.usingPresentationDefinition(options.presentationDefinition)
+    } else if (options.docRequests) {
+      const deviceRequest = DeviceRequest.from('1.0', options.docRequests)
+      deviceResponseBuilder.usingDeviceRequest(deviceRequest)
```

Comment:
Do we transform dcql to device request? Or how does dcql with 18013-7 work?

Reply:
yes exactly